src/main/java/de/tum/in/www1/artemis/service/ProgrammingSubmissionService.java

```diff
@@ -336,10 +337,17 @@ public void notifyUserTriggerBuildForNewSubmissions(Collection<ProgrammingSubmis
      * @param submission ProgrammingSubmission that was just created.
      */
     public void triggerBuildAndNotifyUser(ProgrammingSubmission submission) {
+        final var participation = (ProgrammingExerciseParticipation) submission.getParticipation();
         try {
-            continuousIntegrationService.get().triggerBuild((ProgrammingExerciseParticipation) submission.getParticipation());
+            continuousIntegrationService.get().triggerBuild(participation);
             notifyUserAboutSubmission(submission);
         }
+        catch (BuildPlanNotFoundException e) {
+            if (participation instanceof ProgrammingExerciseStudentParticipation) {
```

Comment:
I don't like this implementation. We know from the participation.initializationState if the build plan still exists. In case the state is inactive, the build plan does not exist and we can avoid running in this exception